### PR TITLE
Feature/8631 WIP

### PR DESCRIPTION
**Pull request by karthik-aot**
*Issue #:* /bcgov/entity/8631

*Description of changes:*

- Add Manual Transaction component
- Unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/util/constants.ts [hunk 1]
```diff
@@ -39,7 +39,8 @@ export enum Role {
   FAS_REPORTS = 'fas_reports',
   FAS_SEARCH = 'fas_search',
   FAS_VIEW = 'fas_view',
-  FAS_CREATE = 'fas_create'
+  FAS_CREATE = 'fas_create',
```
Comment (sumesh-aot) [inline src/util/constants.ts:42]: Added new role `fas_transaction` to enable manual transaction, please update

File: src/components/ViewRoutingSlip/RoutingSlipTransaction.vue [hunk 2]
```diff
@@ -9,32 +9,101 @@
...
         </div>
         <v-btn
-        large
-        color="primary"
-        data-test="btn-add-transaction"
-        v-can:fas_edit.hide
+          large
+          color="primary"
+          data-test="btn-add-transaction"
+          v-can:fas_edit.hide
```
Comment (sumesh-aot) [inline src/components/ViewRoutingSlip/RoutingSlipTransaction.vue:15]: Added new role fas_transaction to enable manual transaction, please update

**Review by sumesh-aot — APPROVED** [review]

File: src/models/RoutingSlip.ts [hunk 3]
```diff
@@ -38,3 +38,11 @@ export interface LinkedRoutingSlips{
   children?: RoutingSlip[],
   parent?: RoutingSlip,
 }
+
+export interface ManualTransactionDetails{
+  quantity?: number,
+  referenceNumber?: string,
+  amount?: number,
+  isFutureEffectiveDate?: boolean,
```
Comment (saravanpa-aot) [inline src/models/RoutingSlip.ts:46]: isFutureEffectiveDate or isFutureEffectiveFilng? Please ask sumesh whats the right term

**Review by saravanpa-aot — APPROVED** [review]

File: src/models/RoutingSlip.ts (hunk 3 above)
Comment (karthik-aot) [reply src/models/RoutingSlip.ts:46]: Confirmed with Sumesh and refactored it to isFutureEffectiveFiling. Thanks

File: src/components/ViewRoutingSlip/AddManualTransactionDetails.vue [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+      addManualTransactionRow,
+      isValid,
+      removeManualTransactionRow
+    }
+  }
+})
+export default class AddManualTransactionDetails extends Vue {
+  @Prop({ default: () => [] }) value: ManualTransactionDetails[]
+
+  public getIndexedTag (tag, index): string {
```
Comment (shabeeb-aot) [inline src/components/ViewRoutingSlip/AddManualTransactionDetails.vue:113]: can we move to composable functions

File: src/composables/ViewRoutingSlip/useAddManualTransactionDetails.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      return value.value
+    },
+    set: (modalValue: ManualTransactionDetails[]) => {
+      context.emit('input', modalValue)
+    }
+  })
+
+  // Input field rules
+  const quantityRules = CommonUtils.requiredFieldRule()
+  const referenceNumberRules = CommonUtils.requiredFieldRule()
```
Comment (shabeeb-aot) [inline src/composables/ViewRoutingSlip/useAddManualTransactionDetails.ts:23]: if both rules are same, may be we can assign to one and re-use it

**Review by shabeeb-aot — APPROVED** [review]

**Pull request merged**